stepconfig/stepconfig.go

```diff
@@ -0,0 +1,169 @@
+package stepconfig
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// ErrInvalidConfig indicates that a configuration is of the wrong type.
+var ErrInvalidConfig = errors.New("conf must be a struct pointer")
+
+// ParseError occurs when a struct field cannot be set.
+type ParseError struct {
+	Field string
+	Value string
+	Err   error
+}
+
+// Error implements builtin errors.Error
+func (e *ParseError) Error() string {
+	segments := []string{e.Field}
+	if e.Value != "" {
+		segments = append(segments, e.Value)
+	}
+	segments = append(segments, e.Err.Error())
+	return strings.Join(segments, ": ")
+}
+
+// Secret variables are not shown in the printed output
+type Secret string
+
+// String implements fmt.Stringer.String.
+// When a Secret is printed, it's masking the underlying string with asterisks.
+func (s Secret) String() string {
+	return strings.Repeat("*", 5)
+}
+
+// Print the name of the struct in blue color followed by a newline,
+// then print all fields and their respective values in separate lines.
+func Print(config interface{}) {
+	v := reflect.ValueOf(config)
+	t := reflect.TypeOf(config)
+
+	log.Infof("%s:\n", t.Name())
+	for i := 0; i < t.NumField(); i++ {
+		fmt.Printf("- %s: %v\n", t.Field(i).Name, v.Field(i).Interface())
```

Comment:
fmt.Printf -> log.Printf, this way no need for newline char and we already using that package